src/doxygen.cpp

```diff
@@ -1049,7 +1049,7 @@ static void addClassToContext(const Entry *root)
       }
     }
     std::unique_ptr<ArgumentList> tArgList;
-    if ((root->lang==SrcLangExt_CSharp || root->lang==SrcLangExt_Java) && (i=fullName.findRev('<'))!=-1)
+    if ((root->lang==SrcLangExt_CSharp || root->lang==SrcLangExt_Java) && (i=fullName.find('<'))!=-1)
     {
```

Comment:
Is this fix also valid for `C#` or are apply there different rules?

Reply:
Honestly, I hoped you would know. I guess the ``findRev`` was there for a reason and this might require to be guarded as Java-only. What I am rather confident about is that this is the proper way to do it for Java. I can guard it if you think that's required.

Reply:
Probably safest is to guard it (I'm not a C# expert either)

Reply:
Unfortunately I am not much familiar with ``C#``. I am aware that it has somewhat similar generics like Java but probably with some differences. I already suspected the issue might apply to ``C#`` as well but do not know for sure. The opinion of someone familiar with ``C#`` would be helpful. After a quick search it appears to me that it is not totally trivial to comprehend the full set of rules from ``C#`` documentation.

Reply:
The latest standard,is to the best of my knowledge, ECMA-334, 5th Edition / December 2017. After this there are changes to the standard as Specification Proposals but they have not bee officially standardized (see e.g. https://en.wikipedia.org/wiki/C_Sharp_(programming_language)).